osu.Game/Audio/EBUr128LoudnessNormalization.cs

```diff
@@ -0,0 +1,325 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using ManagedBass;
+using osu.Framework.Audio.Callbacks;
+using osu.Framework.Audio.Track;
+
+namespace osu.Game.Audio
+{
+    public class EbUr128LoudnessNormalization
```

Comment:
This is implemented in bass already. Is there a reason for not using the plugin and reimplementing from scratch?

If there's a valid reason (ie. platform availability) then this class needs to be 100% documented, including links to reference papers showing where formulae came from.

Reply:
The reason for not using a bass plugin is simply that when I looked for it the only thing I found was https://www.nuget.org/packages/ManagedBass.ReplayGain/ and this didn't seem reliable. I took a look at the branch you linked in the other reply but it seems like the actual calls to bass are in a local branch of the framework. If you could provide a link to the plugin you mentioned and/or to the framework branch that'd be great cause at the end of the day using bass directly would be better for sure.

Reply:
Unfortunately I can't find the branch I had for that 😦 . I don't recall using that package, but I may have also hacked together a local implementation.

It's fine to implement ourselves, but it does need to be documented as I stated. During review and future perusal of the code it should be very easy for a developer to cross-check against a known implementation, and anything non-obvious should be commented inline.